lib/tasks/generate-from-blueprint.js

```diff
@@ -37,7 +56,7 @@ class GenerateTask extends Task {
 
     // otherwise, use default addon-import
     if (noAddonBlueprint.indexOf(name) < 0 && !addonBlueprint && options.args[1]) {
-      let mainBlueprintSupportsAddon = mainBlueprint && mainBlueprint.supportsAddon();
+      let mainBlueprintSupportsAddon = mainBlueprint && mainBlueprint.supportsAddon(blueprintOptions);
```

Comment:
This feels slightly weird. I was also thinking about passing the options to the blueprint upon instantiation, but that's a bit more involved. The options would need to be passed to the `Blueprint.lookup` method and so on.

Reply:
I just noticed that https://github.com/ember-cli/ember-cli/issues/5481 is a similar issue. I will update this PR to pass along the blueprint options and project upon creation of the blueprint instance.

Reply:
Done!